frontend_v2/src/app/components/publiclists/challengelist/challengelist.component.ts

```diff
@@ -334,4 +372,32 @@ export class ChallengelistComponent implements OnInit {
       () => {}
     );
   }
+
+  /**
+   * Fetch challenges from backend.
+   * @param path  Challenge fetch URL
+   * @param callback  Callback Function.
+   */
+  fetchParticipatedChallengesFromApi(path, callback = null) {
+    const SELF = this;
+    SELF.apiService.getUrl(path, true, false).subscribe(
+      data => {
+        if (path.endsWith('future')) {
+          SELF.upcomingChallenges = data['results'];
+        } else if (path.endsWith('present')) {
+          SELF.ongoingChallenges = data['results'];
+        } else if (path.endsWith('past')) {
+          SELF.pastChallenges = data['results'];
+        }
+        SELF.filteredChallenges = SELF.upcomingChallenges.concat(SELF.ongoingChallenges, SELF.pastChallenges);
+        SELF.filteredOngoingChallenges = SELF.ongoingChallenges;
```

Comment:
As we are using same variable name in hosted challenges API also so do you think we should name different variable for participated challenges and the hosted challenges?

Reply:
@Sanji515 we can name the different variables for participated and hosted challenges but i think it's fine this way also 'coz the variable will be assigned each time the required url is visited.